src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -1047,6 +1067,99 @@ private void EndStartupOperation(ScriptHostStartupOperation operation)
             _logger.StartupOperationCompleted(operation.Id);
         }
 
+        private async Task LogErrorWithTransientOtelLogger(string log)
+        {
+            var loggerFactory = BuildOtelLoggerFactory();
+
+            if (loggerFactory is not null)
+            {
+                var logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
+                logger.LogError(log);
+
+                // Delay increases the chance that the log is sent to AppInsights/Otel before the logger factory is disposed
+                await Task.Delay(2000);
+                // Do a force flush as the host is shutting down and we want to ensure the logs are sent before disposing the logger factory
+                ForceFlush(loggerFactory);
+                // Give some time for the logger to flush
+                await Task.Delay(4000);
+            }
+        }
+
+        private ILoggerFactory BuildOtelLoggerFactory()
```

Comment:
I see we are using OTel only. I get that using AppInsights SDK isn't critical for integrating with customer code, because that won't be loaded here. But I wonder if there will be difference in the schema at all between OTel and AppInsights SDK?

By not using AppInsights SDK for non-otel users, will we be emitting an inconsistent schema here?